mesheryctl/pkg/utils/helpers.go

```diff
@@ -158,18 +158,55 @@ func SetFileLocation() error {
 
 //PreReqCheck prerequisites check
 func PreReqCheck() error {
+	//Check whether docker daemon is running or not
+	if err := exec.Command("docker", "ps").Run(); err != nil {
+		log.Info("Docker daemon is not running")
+		//No auto installation of docker for windows
+		if runtime.GOOS == "windows" {
+			return errors.Wrap(err, "please start docker daemon. Run `masheryctl system start` once docker is started.")
+		}
+		err = startdockerdaemon()
+		if err != nil {
+			return errors.Wrap(err, "failed to start docker daemon. Run `masheryctl system start` once docker is started.")
+		}
+	}
 	//Check for installed docker-compose on client system
 	if err := exec.Command("docker-compose", "-v").Run(); err != nil {
 		log.Info("Docker-Compose is not installed")
 		//No auto installation of Docker-compose for windows
 		if runtime.GOOS == "windows" {
-			return errors.Wrap(err, "please install docker-compose")
+			return errors.Wrap(err, "please install docker-compose. Run `masheryctl system start` once docker-compose is installed.")
 		}
 		err = installprereq()
 		if err != nil {
-			return errors.Wrap(err, "failed to install prerequisites")
+			return errors.Wrap(err, "failed to install prerequisites. Run `masheryctl system start` once docker-compose is installed.")
+		}
+	}
+	return nil
+}
+
+func startdockerdaemon() error {
+	log.Info("Attempting to start docker daemon")
+	// TODO: read user input on whether to start docker or not.
+	// once user gaves permission, start docker daemon on linux/macOS
+	if runtime.GOOS == "linux" {
+		if err := exec.Command("service", "docker", "start").Run(); err != nil {
+			return errors.Wrap(err, "Please start docker then run the command `mesheryctl system start` to start mesheryctl")
+		}
+	} else {
+		// Assuming we are on macOS, try to start Docker from default path
+		cmd := exec.Command("/Applications/Docker.app/Contents/MacOS/Docker")
+		err := cmd.Start()
+		if err != nil {
+			return errors.Wrap(err, "Please start docker then run the command `mesheryctl system start` to start mesheryctl")
+		}
+		// wait for few seconds for docker to start
+		err = exec.Command("sleep", "20").Run()
```

Comment:
I'm not sure if there's a more clever way to detect if the Docker daemon is running other than sleeping. If there is and it's easy to implement, that might be nice. If not, no problem.

Reply:
`systemctl is-active docker` this command works in Linux, @hexxdump You can use this in macOS and if the status is active you can proceed. Again I am not sure if it is efficient., Feel free to choose which way is more efficient according to you.

Reply:
@NupurThakur27 I already checked, If i could use "systemctl start docker" for both linux and MacOS. However, systemctl is not available on MacOS. Hence I had to write different logic for each OS. 
I did not like the idea of the sleep command myself, but since its the install flow, I thought the sleep statement(delay) could be justifiable.

Reply:
Okay @hexxdump, Then it makes sense.

Reply:
Looks good to me. Question: Could we increase the timeout if it fails to start within this time? And then fail hard if it does not start with the increased timeout ?